packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts

```diff
@@ -62,14 +69,23 @@ export class InsuredBridgeL2Client {
     // Define a config to bound the queries by.
     const blockSearchConfig = {
       fromBlock: this.firstBlockToSearch,
-      toBlock: this.endingBlockNumber || (await this.l2Web3.eth.getBlockNumber()),
```

Comment:
why did you need to make this change? did accessing the `endingBlockNumber` not work? also I don't see why you need to re-declare the `lastBlockToSearch`

Reply:
So i discovered this bug when I passed an `endingBlockNumber` into this function. You'll notice that after each loop, the `fromBlock` gets set to the `toBlock + 1`, but the `toBlock` does not get reset, so after the first loop, the `toBlock` remains at the same value while `fromBlock` is now `> toBlock`.

Reply:
So you'll notice that after each loop, `fromBlock` gets set to `toBlock + 1`, but `toBlock` is not reset. So this causes the next loop to have `fromBlock > toBlock`. Shouldn't we be incrementing `toBlock` as well? Or perhaps we should rename the constructor variable `endingBlockNumber` to `blockSearchWindowLength`, wdyt?